test/e2e/specs/specs.go

```diff
@@ -229,11 +233,33 @@ func (t *TestPod) VerifyExecInPodSucceedWithFullOutput(f *framework.Framework, c
 
 // VerifyExecInPodFail verifies shell cmd in target pod fail with certain exit code.
 func (t *TestPod) VerifyExecInPodFail(f *framework.Framework, containerName, shExec string, exitCode int) {
-	stdout, stderr, err := e2epod.ExecCommandInContainerWithFullOutput(f, t.pod.Name, containerName, "/bin/sh", "-c", shExec)
+	stdout, stderr, err := execCommandInContainerWithFullOutputWithRetry(f, t.pod.Name, containerName, "/bin/sh", "-c", shExec)
 	gomega.Expect(err).Should(gomega.HaveOccurred(),
 		fmt.Sprintf("%q should fail with exit code %d, but exit without error\nstdout: %s\nstderr: %s", shExec, exitCode, stdout, stderr))
 }
 
+// execCommandInContainerWithFullOutputWithRetry executes a command in a target pod and retries every 2 minutes until timeout(10 min) or success.
+func execCommandInContainerWithFullOutputWithRetry(f *framework.Framework, podName, containerName string, cmd ...string) (string, string, error) {
```

Comment:
> Lets use backoff mechanism for retrying,  gradually increasing the time between retries. It is a good way to handle  transient errors. The manual retry loop with fixed interval is less efficient  because maybe it succeeds shortly after failing the first time, but then we would have to wait a whole 2 min. Here's an example  https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/blob/68f4228cd743b55df3779f2f149f50866c9049de/pkg/sidecar_mounter/sidecar_mounter.go#L416-L451